### ISSUE TITLE
Notifications?

### ISSUE THREAD

**Issue by bwiernik** [akaplo/NativeHangouts#1]
Any chance of adding support for OS X native notifications?

**Comment by akaplo** [comment]
Highly doubtful.

A quick search for displaying notifications from a WebView returns pages such as [this StackOverflow article](http://stackoverflow.com/questions/14237086/how-to-handle-html5-web-notifications-with-a-cocoa-webview), which, while aged, tell me that it'll be extremely hard to pull notifications from it.  If you know anything about it and could contribute, it'd be more than welcome!  

As the state of this repository suggests, I wrote this quickly and never thought about it again (although I do use it every day...very well worth the time I spent writing it :stuck_out_tongue:.  I'm considering extending it to support opening a file picker for adding images to your conversations soon, FYI).

**Comment by bwiernik** [comment]
Thanks, thought that would be the case. Sorry I can't help more; my JS skills aren't the sharpest. Though I did run across this project that might be useful: https://github.com/MacGapProject/MacGap1

**Comment by irae** [comment]
Both wrapper for WhatsApp and for Facebook do notifications and badges. They inject some JavaScript and talk back to the application, and the application dispatches system messages. Here are the projects you can use as inspiration. If I have some time I might even do a PR for that.

For WhatsApp (formerly called WhatsMac): https://github.com/stonesam92/ChitChat
For Facebook Messenger: http://fbmacmessenger.rsms.me/

Both are pretty stable, and both support Sparkle, for auto updates.

**Comment by akaplo** [comment]
You're right, @irae.  I've noticed that Slack's desktop app (while I believe it's an Electron app) does something like this too.  

Let me know if you start work on this; otherwise, I might give it a go next week after my finals end.

**Comment by irae** [comment]
I have not used Slack in a while, but I did use it between Mountain Lion and Yosemite transition and I reported bugs to them regarding the Safari version. Since Electron uses V8 and the Chromium rendering engine, I believe Slack is not Electron based and uses WKWebView or the legacy NSWebView (or whatever it is called on the mac). I am unlikely to work on this soon. I am using Fluid right now and kinda works for me, although no notifications are available.
